keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,919 @@
+            specified if `include_top` is True, and if no `weights` argument is
+            specified.
+        :param classifier_activation: A `str` or callable. The activation function to
+            use on the "top" layer. Ignored unless `include_top=True`. Set
+            `classifier_activation=None` to return the logits of the "top" layer.
+        :param include_preprocessing: Boolean, whether to include the preprocessing layer
+            (`Rescaling`) at the bottom of the network. Defaults to `True`.
+    Returns:
+        A `tf.keras.Model` instance.
+    Raises:
+        ValueError: in case of invalid argument for `weights`, or invalid input
+            shape.
+        ValueError: if `classifier_activation` is not `softmax` or `None` when
+            using a pretrained top layer.
+    """
+    # Validate parameters
+    available_weight_variants = DEPTH_TO_WEIGHT_VARIANTS[depth]
+    if weights == "imagenet":
+        max_input_shape = max(available_weight_variants)
+        """logging.warning(
+            f"Received `imagenet` argument without "
+            f"explicit weights input size. Picking weights trained with "
+            f"biggest available shape: imagenet-i{max_input_shape}"
+        )"""
+        weights = f"{weights}-i{max_input_shape}"
+
+    weights_allow_list = [f"imagenet-i{x}" for x in available_weight_variants]
+    if not (weights in {*weights_allow_list, None} or file_io.file_exists_v2(weights)):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` (random initialization), `'imagenet'` "
+            "(pre-training on ImageNet, with highest available input shape),"
+            " or the path to the weights file to be loaded. "
+            f"For ResNetRS{depth} the following weight variants are "
+            f"available {weights_allow_list} (default=highest)."
+            f" Received weights={weights}"
+        )
+
+    if weights in weights_allow_list and include_top and classes != 1000:
+        raise ValueError(
+            f"If using `weights` as `'imagenet'` or any of {weights_allow_list} with "
+            f"`include_top` as true, `classes` should be 1000. "
+            f"Received classes={classes}"
+        )
+    input_shape = imagenet_utils.obtain_input_shape(
+      input_shape,
+      default_size=224,
+      min_size=32,
+      data_format=backend.image_data_format(),
+      require_flatten=include_top,
+      weights=weights)
+    # Define input tensor
+    if input_tensor is None:
+        img_input = layers.Input(shape=input_shape)
+    else:
+        if not backend.is_keras_tensor(input_tensor):
+            img_input = layers.Input(tensor=input_tensor, shape=input_shape)
+        else:
+            img_input = input_tensor
+    bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+    x = img_input
+    if include_preprocessing:
+        x = layers.Rescaling(scale=1. / 255)(x)
```

Comment:
Did you train the model from scratch to match this preprocessing scheme? Or were the weights extracted from somewhere?

Reply:
@fchollet The weights were converted by me, from the [original repository](https://github.com/tensorflow/tpu/tree/master/models/official/resnet/resnet_rs). The preprocessing layer should be:
```python
x = Normalization(
    mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
    variance=[(0.229 * 255) ** 2, (0.224 * 255) ** 2, (0.225 * 255) ** 2],
)(x)
``` 
to match the original preprocessing.

Reply:
Weights were converted by @sebastian-sz  👍 

I have also updated preprocessing layer.

Below you can find results for imagenet evaluation, 

<!DOCTYPE html>
Depth | Image size | Official Top 1 | With converted weights Top 1 | With converted weights Top 5 | Top 1 diff
-- | -- | -- | -- | -- | --
50 | 160 | 78.8 | 78.9 | 94.3 | +0.1
101 | 160 | 80.3 | 80.3 | 95.0 | 0.0
101 | 192 | 81.2 | 81.3 | 95.6 | +0.1
152 | 192 | 82.0 | 82.1 | 95.9 | +0.1
152 | 224 | 82.2 | 82.9 | 96.2 | +0.7
152 | 256 | 83.0 | 83.0 | 96.3 | 0.0
200 | 256 | 83.4 | 83.4 | 96.5 | 0.0
270 | 256 | 83.8 | 83.8 | 96.7 | 0.0
350 | 256 | 84.0 | 84.0 | 96.8 | 0.0
350 | 320 | 84.2 | 84.3 | 96.8 | +0.1
420 | 320 | 84.4 | 84.4 | 96.8 | 0.0

[Evaluation script](https://github.com/sebastian-sz/resnet-rs-keras/tree/main/imagenet_evaluation)

Reply:
> Following are comparative imagenet evaluation results with official weights,

Thanks for the info! Did you generate these numbers from the latest code in this PR? Do you have a Colab I can check out?

Reply:
Those were earlier generated numbers.

Please refer below Colab link, where I have ran evaluation script on latest code from this PR.  Numbers are approximately matching with official repo.

https://colab.research.google.com/drive/1XoR18LoPhZ0Y1T3GXUdXvSmYlkJhSCS9?usp=sharing